proposals/0000-midstream-bindists.md

```diff
@@ -0,0 +1,199 @@
+# Midstream bindists
+
+## Abstract
+
+Bindists for the Haskell toolchain have been produced by upstream (the developers of each respective tool) for
+a long time and many tools rely on these "official" bindists (e.g. GHCup and stack).
+
+We propose here that bindists are additionally built and maintained in joint effort by the projects that
+provide installation experiences, removing the hard dependency on upstream bindists entirely.
+
+## Background
+
+Historically, installers like GHCup and stack have used upstream bindists for mainly one reason: it's easy to do
+so and doesn't require further efforts.
+
+However, using upstream bindists directly is extremely rare in the Linux world of distribution. Most distributions
+build, package, test and curate binary packages themselves, not only because they have custom formats, but for
+reasons of control, trust and quality.
+
+The relationship of GHCup and bindists has also been described in the blog post
+[GHCup is not an installer](https://hasufell.github.io/posts/2023-11-14-ghcup-is-not-an-installer.html) recently.
+
+## Problem Statement
+
+From the perspective of a GHCup developer, there are several issues with relying on upstream bindists.
+
+### Platform support
+
+GHC and other tools have in the past dropped support for certain platforms either entirely or requested
+the community to step up and do the work (e.g. on GHC CI).
+
+E.g. [GHC ARMv7 support was dropped silently without any call for help](https://gitlab.haskell.org/ghc/ghc/-/issues/21177#note_470440). Similarly, FreeBSD support just ceased to exist when the GHC FreeBSD CI stopped working. Later the community asked for a [revival](https://gitlab.haskell.org/groups/ghc/-/epics/5), but nothing signifcant has happened so far.
```

Comment:
The linked issue notes three (maybe four?) people who are interested, none of whom have cared enough to follow up except for @hasufell . My perception has been that the person who mostly wants FreeBSD support is @hasufell , and this doesn't convince me that there's much community demand.

Reply:
This is a chicken and egg problem: is the demand low, because the tooling is totally busted and end-users just use a linux VM?

Are windows users at 25%, because the tooling experience is great or because they switch to WSL and don't bother?

Why don't we just drop windows?

Again: I do not subscribe to this approach. We clearly have different goals/perceptions here.